LinkedList/LinkedList/TSLList.h

```diff
@@ -0,0 +1,350 @@
+
+	FIterator end()
+	{
+		return FIterator(nullptr);
+	}
+
+	//-------------------------Access
+	Type& operator[] (const int Index)
+	{
+		FNode* Temporal = Head; 
+		for (int i = 0; i < Index; i++)
+		{
+			Temporal = Temporal->Next;
+		}
+		return Temporal->Element;
+	}
+
+	const Type& operator[] (const int Index) const
+	{
+		FNode* Temporal = Head;
+		for (int i = 0; i < Index; i++)
+		{
+			Temporal = Temporal->Next;
+		}
+		return Temporal->Element;
+	}
+
+	//-----------------------------------------------
+	TSLList& operator= (const TSLList<Type>& OtherTSLlist)
+	{
+		Clear();
+		FNode* CurrentNode = OtherTSLlist.Head;
+
+		for (int i = 0; i < Size; i++)
+		{
+			AddTail(CurrentNode->Element);
+			CurrentNode = CurrentNode->Next;
+		}
+		return *this;
+
+	}
+
+	//-----------------------------------------------
+	Type& GetHead()
+	{
+		return Head->Element;
+	}
+
+	const Type& GetHead() const
+	{
+		return Head->Element;
+	}
+
+	//-----------------------------------------------
+	Type& GetTail()
+	{
+		return Tail->Element;
+	}
+
+	const Type& GetTail() const
+	{
+		return Tail->Element;
+	}
+
+	//---------------------------------Size
+	bool IsEmpty()
+	{
+		return Size == 0;
+	}
+
+	//------------------------------------------------
+	int GetSize()
+	{
+		return Size;
+	}
+
+	//-----------------------------------Modifiers
+	void AddHead(const Type& InHead)
+	{
+		FNode* NewHead = new FNode;
+		NewHead->Element = InHead;
+
+		if (Head == nullptr)
+		{
+			Head = NewHead;
+			Tail = NewHead;
+			Size++;
+		}
+		else
+		{
+			NewHead->Next = Head;
+			Head = NewHead;
+			Size++;
+		}
+	}
+
+	//-------------------------------------------------
+	void AddTail(const Type& InTail)
+	{
+
+		if (Head == nullptr)
+		{
+			AddHead(InTail);
+		}
+		else
+		{
+			Tail->Next = new FNode;
+			Tail->Next->Element = InTail;
+			Tail = Tail->Next;
+			Size++;
+		}
+	}
+	
+	//----------------------------------------------------
+	void Insert(const Type& NewItem, const int Index)
+	{
+		if (Index == 0)
+		{
+			AddHead(NewItem);
+		}
+		else if(Index == Size)
+		{
+			AddTail(NewItem);
+		}
+
+		else if (Index > 0 && Index < Size)
+		{
+			FNode* Temporal = Head;
+			FNode* Container = new FNode;
+
+			for (int i = 0; i < Index; i++)
+			{
+				Temporal = Temporal->Next;
+			}
+			Container->Element = Temporal->Element;
+
+			Temporal->Element = NewItem;
+			Container->Next = Temporal->Next;
+			Temporal->Next = Container;
+			Size++;
+		}
+	}
+	//---------------------------------------------------------
+	void RemoveHead()
+	{
+		if (Size != 0)
+		{
+			FNode* NewHead = Head->Next;
+			delete Head;
+			Head = NewHead;
+			--Size;
+		}
+
+		else if (Size == 1)
+		{
+			delete Head;
+			Head = nullptr;
+			Tail = nullptr;
+			--Size;
+		}
+	}
+
+	//---------------------------------------------------------
+	void Remove(const int Index)
+	{
+		FNode* Temporal = Head;
+
+		if (Index == 0)
+		{
+			RemoveHead();
+		}
+
+		else if (Index >= 0 && Index <= Size)
```

Comment:
This could be Index > 0 since you're already considering that case

Reply:
Also, same thing about indexes outta' bounds, it'd be nice if that was the first consideration of your function

Reply:
It would also help with some readability. For even though else ifs work, this is not the best use for them. On conditions like these if-else statements are good enough

Reply:
Another thing, Size should be out of bounds here